### PR TITLE
Add: DeepSeek V4 Pro staged EP2 execution smoke

### PR DESCRIPTION
**Pull request by lwDavid**
- Add execution-only golden-runner output capture with memory-bounded
  resident tensor uploads and configurable distributed startup deadlines.
- Add staged 61-layer decode and prefill drivers that reuse four compiled
  layer specializations, chain recurrent state across rounds, and run the
  final normalized-hidden-state tail.
- Add low-memory zero-INT8 expert fixtures and stable per-layer synthetic
  initialization across cache overrides.
- Document the EP2 smoke scope, no-fusion task-submit workflow, reduced
  expert topology, and current lack of native MXFP8-MXFP4 or numerical
  validation.

Depends on hw-native-sys/pypto#2340.

### CODE REVIEW

File: models/deepseek_v4_pro/staged_fwd_tail.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    import torch
+    from golden import TensorSpec
+
+    if token_count not in (DECODE_TOKENS, PREFILL_TOKENS):
+        raise ValueError(
+            f"tail token count must be decode ({DECODE_TOKENS}) or prefill "
+            f"({PREFILL_TOKENS}), got {token_count}"
+        )
+
+    base = [5.9166, -3.6223, -2.9324, -3.3124]
```
[the remark this answers is not in the document]
Comment (lwDavid) [reply models/deepseek_v4_pro/staged_fwd_tail.py:105]: Fixed in cfed2b7. The tail fixture now validates that the base entry count matches `HC_MULT` and raises a clear configuration error otherwise.

File: models/deepseek_v4_pro/prefill_layer.py [hunk 2]
```diff
@@ -1299,6 +1304,8 @@ def cmp(actual, expected, **kwargs):
                         help="Comma-separated per-request logical chunk lengths.")
     parser.add_argument("--start-positions", type=str, default=None,
                         help="Comma-separated per-request prior context lengths; defaults to all zeros.")
+    parser.add_argument("--smoke-weights", action="store_true", default=False,
+                        help="use lazy constant expert weights for execution-only smoke runs")
```
[the remark this answers is not in the document]
Comment (lwDavid) [reply models/deepseek_v4_pro/prefill_layer.py:1308]: Fixed in cfed2b7. Prefill smoke weights now disable the golden callback and calibrated comparison map, while normal runs retain the existing validation.

File: models/deepseek_v4_pro/decode_layer.py [hunk 3]
```diff
@@ -799,6 +816,8 @@ def init_input_ids():
     parser.add_argument("--start-pos", type=int, default=DECODE_START_POS,
                         help="Fixture-only start_pos for all batches; default is the 8k target position.")
     parser.add_argument("--layer-id", type=int, default=10)
+    parser.add_argument("--smoke-weights", action="store_true", default=False,
+                        help="use lazy constant expert weights for execution-only smoke runs")
```
[the remark this answers is not in the document]
Comment (lwDavid) [reply models/deepseek_v4_pro/decode_layer.py:820]: Fixed in cfed2b7. Decode smoke weights now disable the golden callback and calibrated comparison map, while normal runs retain the existing validation.

File: golden/runner.py [hunk 4]
```diff
@@ -746,23 +760,93 @@ def _try_l3_dispatch(
...
+
+    from pypto.runtime import DistributedWorker
+
+    config = run_config if bench else None
+    return DistributedWorker(
+        compiled,
+        config,
+        inherited_host_tensors=inherited_host_tensors,
+        **kwargs,
+    )
```
[the remark this answers is not in the document]
Comment (lwDavid) [reply golden/runner.py:861]: Fixed in cfed2b7. The L3 helper now inspects each target signature and forwards `startup_timeout_s` only when supported. Tests cover legacy prepare plus legacy and current fallback workers.

File: models/deepseek_v4_pro/moe.py [hunk 5]
```diff
@@ -980,6 +999,8 @@ def init_input_ids():
                         help=f"active token count for MoE dispatch/combine (0..{T})")
     parser.add_argument("--balanced-routing", action="store_true", default=False,
                         help="use deterministic hash routes balanced evenly across all experts")
+    parser.add_argument("--smoke-weights", action="store_true", default=False,
+                        help="use lazy constant expert weights for execution-only smoke runs")
```
[the remark this answers is not in the document]
Comment (lwDavid) [reply models/deepseek_v4_pro/moe.py:1003]: Fixed in cfed2b7. MoE smoke weights now disable the golden callback, cached golden data, and calibrated comparison map. Normal validation is unchanged.

**Comment by lwDavid** [comment]
This PR is not needed due to test resource restrictions.
